### DISCUSSION TITLE
Not getting a warning if closing an SQL Script tab without saving

### DISCUSSION THREAD

**Discussion by MikeCSanJose** [dbeaver/dbeaver#35158] [Q&A] [answered]
### Description

I just lost a small snippet of code because I closed the tab in the editor and did not get a warning that it was not saved.  I thought there was a warning in previous releases.

### DBeaver Version

Version 24.1.4.202408041450

### Operating System

Windows 10 Enterprise Version:22H2 Build:19045.4651

### Database and driver

Driver: Snowflake

### Steps to reproduce

1. Open a New SQL Script
2. type anything
3. close the tab

**Comment by E1izabeth** [comment, marked as answer]
Since DBeaver 24.1.1 the setting `Auto-save editor on close` is turned on by default.
You can turn it off in `Preferences->Editors->SQL Editor`
![image](https://github.com/user-attachments/assets/9578f26d-11f5-49b7-9e3d-18b7b28d1863)

**Reply by MikeCSanJose** [reply]
Thanks.  I'll un-select that.  Auto-save gets me into bad habits, when other apps don't do the same.  Also one of my habits - I will often go in and play around with some edits and exit, expecting them not to be saved.  I think that might explain some issues I've had recently where my code must have gotten unintentionally saved when I had no expectation that it would have been.  I guess I need to spend more time reading the release notes.  One of the main reasons I'm using DBeaver for Snowflake is that Snowsight autosaves continuously - which is horrible for my way of working
